app/controllers/food_controller.rb

```diff
@@ -0,0 +1,13 @@
+class FoodController < ApplicationController
+  def index
+    @foods = FoodFacade.search(params[:q])
+
+    if @foods == []
+      flash[:error] = 'No food found'
+      render 'welcome/index'
+    else
+      @foods
```

Comment:
No need to call this `@foods` explicitly here. Once you set it, it's available in the view.